pytest_plugins/logging_hooks.py

```diff
@@ -0,0 +1,69 @@
+import logging
+
+import logzero
+import pytest
+from xdist import is_xdist_worker
+
+from robottelo.logging import DEFAULT_DATE_FORMAT
+from robottelo.logging import logger
+from robottelo.logging import robottelo_log_dir
+from robottelo.logging import robottelo_log_file
+
+try:
+    from pytest_reportportal import RPLogger
+    from pytest_reportportal import RPLogHandler
+except ImportError:
+    pass
+
+
+@pytest.fixture(autouse=True, scope='session')
+def configure_logging(request, worker_id):
+    """Handle xdist and ReportPortal logging configuration at session start
+
+    Set up a separate logger for each pytest-xdist worker
+    if worker_id != 'master' then xdist is running in multi-threading so
+    a logfile named 'robottelo_gw{worker_id}.log' will be created.
+
+    Add a handler for ReportPortal logging
+    """
+    worker_formatter = logzero.LogFormatter(
+        fmt=f'%(asctime)s - {worker_id} - %(name)s - %(levelname)s - %(message)s',
+        datefmt=DEFAULT_DATE_FORMAT,
+    )
+    use_rp_logger = hasattr(request.node.config, 'py_test_service')
+    if use_rp_logger:
+        logging.setLoggerClass(RPLogger)
+
+    if is_xdist_worker(request):
+        if f'{worker_id}' not in [h.get_name() for h in logger.handlers]:
+            # Track the core logger's file handler level, set it in case core logger wasn't set
+            worker_log_level = 'INFO'
+            handlers_to_remove = [
+                h
+                for h in logger.handlers
+                if isinstance(h, logging.FileHandler)
+                and getattr(h, 'baseFilename', None) == str(robottelo_log_file)
+            ]
+            for handler in handlers_to_remove:
+                logger.removeHandler(handler)
+                worker_log_level = handler.level
+            worker_handler = logging.FileHandler(
+                robottelo_log_dir.joinpath(f'robottelo_{worker_id}.log')
+            )
+            worker_handler.set_name(f'{worker_id}')
+            worker_handler.setFormatter(worker_formatter)
+            worker_handler.setLevel(worker_log_level)
+            logger.addHandler(worker_handler)
+
+            if use_rp_logger:
+                rp_handler = RPLogHandler(request.node.config.py_test_service)
+                rp_handler.setFormatter(worker_formatter)
+                # logger.addHandler(rp_handler)
```

Comment:
Yeah, so, weird right?  Well, I don't quite understand _why_, but RP's own documentation never adds the handler to the logger.  Calling `logging.setLoggerClass(RPLogger)` up above injects a hook into the core logger that forces RPLogHandler.emit to get called, even without adding the handler to the logger. Why they did this instead of just letting people control the handlers.... I don't know.

Reply:
It also appears to completely ignore the formatter that is set on the `RPLogHandler` instance, and `RPLogHandler.emit` only emits the message, unformatted. This is no change in behavior compared to the current RP logging hook in robottelo.